[jit] Remove forward compat code for serialization format

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25440 [jit] Remove forward compat code for serialization format**

See the comments deleted for what this PR is all about

Differential Revision: [D17125690](https://our.internmc.facebook.com/intern/diff/D17125690)

suo committed:
[jit] Remove forward compat code for serialization format

See the comments deleted for what this PR is all about

suo committed:
Update on "[jit] Remove forward compat code for serialization format"


See the comments deleted for what this PR is all about

suo committed:
Update on "[jit] Remove forward compat code for serialization format"


See the comments deleted for what this PR is all about

Differential Revision: [D17125690](https://our.internmc.facebook.com/intern/diff/D17125690)

zdevito reviewed (approved):
I auto-approve diffs that have this many deletions. Hopefully it won't break too bad :)

suo committed:
Update on "[jit] Remove forward compat code for serialization format"


See the comments deleted for what this PR is all about

Differential Revision: [D17125690](https://our.internmc.facebook.com/intern/diff/D17125690)

suo committed:
Update on "[jit] Remove forward compat code for serialization format"


See the comments deleted for what this PR is all about

Differential Revision: [D17125690](https://our.internmc.facebook.com/intern/diff/D17125690)
